modules/cas_cache/layer_cache_management.c

```diff
@@ -2353,12 +2353,27 @@ int cache_mngt_activate(struct ocf_mngt_cache_standby_activate_config *cfg,
 	if (result == -KCAS_ERR_WAITING_INTERRUPTED)
 		goto out_cache_put;
 
-	cas_lazy_thread_stop(context->rollback_thread);
+	result = _cache_start_finalize(cache, -1, true);
+	if (result)
+        goto err;
 
+    cas_lazy_thread_stop(context->rollback_thread);
 	if (result)
 		goto err_free_context;
 
-	result = _cache_start_finalize(cache, -1, true);
+err:
+    _cache_mngt_async_context_init(&context->async);
+    ocf_mngt_cache_stop(cache, _cache_mngt_cache_stop_rollback_complete,
+                    context);
+    rollback_result = wait_for_completion_interruptible(&context->async.cmpl);
+
+    rollback_result = _cache_mngt_async_caller_set_result(&context->async,
+                    rollback_result);
+    
+    if (rollback_result != -KCAS_ERR_WAITING_INTERRUPTED)
+        kfree(context);
+    
+    return result;
 
 err_free_context:
```

Comment:
Personally, I would add a second label here (i.e. `finish`, `end` or `our_success`) because `err_free_context` suggests that an error occurred which is not true if the code jumpes here after a successful activation. Having two labels will increase readability